Assets/Scripts/PlayerFall.cs

```diff
@@ -15,40 +15,46 @@ public class PlayerFall : MonoBehaviour
     private float desiredCameraHeight = 5.0f;
     private Vector3 direction;
     private float timer = 0.0f;
-    private float delayuntilFastFall = 1.0f;
-    //falling audio
+    private float delayuntilFastFall = .5f;
+    
+    //falling audio at this time overriden in BeginFall method
     [SerializeField]
     private AudioSource fallingAudio;
-    //flags
-    private bool AudioPlayed = false;
+    
+    //flag
     public bool trapOpened {get; set;}
-    private bool CameraChanged = false;
     
-    void Awake(){
+    void Start(){
+        
+        GameObject.Find("TrapDoor").transform.GetChild(0).transform.GetChild(0).GetComponent<TrapDoorOpen>().TrapOpened += BeginFall;
+        GameObject.Find("TrapDoor").transform.GetChild(1).transform.GetChild(0).GetComponent<TrapDoorOpen>().TrapOpened += BeginFall;
+    }
+    
+    void BeginFall(){
         direction = new Vector3(0, desiredCameraHeight, 0);
         Player = GameObject.Find("MainPlayer");
         PlayerBod = Player.GetComponent<Rigidbody>();
-        fallingAudio = GetComponent<AudioSource>();
-        //we must remove the freeze y position constraint
-        PlayerBod.constraints = RigidbodyConstraints.FreezeRotation;
-        
+        fallingAudio = GameObject.FindWithTag("Boss").GetComponents<AudioSource>()[1];
+        trapOpened = true;
+        ChangeCamera();
+
     }
     
     void Update(){
         
         if(trapOpened){

```

Comment:
What's the point of having this if you added the events for when the trap doors open? I would just get rid of the update loop and put that logic inside a function that isn't getting called every frame.

Reply:
There is a point as I want to change position of player every frame as well as increase volume every frame.